src/components/AttachmentCarousel/index.js

```diff
@@ -0,0 +1,248 @@
+
+        return {
+            source,
+            file,
+        };
+    }
+
+    /**
+     * Toggles the visibility of the arrows
+     * @param {Boolean} shouldShowArrow
+     */
+    toggleArrowsVisibility(shouldShowArrow) {
+        this.setState({shouldShowArrow});
+    }
+
+    /**
+     * This is called when there are new reports to set the state
+     */
+    makeStateWithReports() {
+        let page;
+        const actions = ReportActionsUtils.getSortedReportActions(_.values(this.props.reportActions), true);
+
+        /**
+         * Looping to filter out attachments and retrieve the src URL and name of attachments.
+         */
+        const attachments = [];
+        _.forEach(actions, ({originalMessage, message}) => {
+            // Check for attachment which hasn't been deleted
+            if (!originalMessage || !originalMessage.html || _.some(message, m => m.isEdited)) {
+                return;
+            }
+            const matches = [...originalMessage.html.matchAll(CONST.REGEX.ATTACHMENT_DATA)];
+
+            // matchAll captured both source url and name of the attachment
+            if (matches.length === 2) {
+                const [originalSource, name] = _.map(matches, m => m[2]);
+
+                // Update the image URL so the images can be accessed depending on the config environment.
+                // Eg: while using Ngrok the image path is from an Ngrok URL and not an Expensify URL.
+                const source = tryResolveUrlFromApiRoot(originalSource);
+                if (source === this.state.source) {
+                    page = attachments.length;
+                }
+
+                attachments.push({source, file: {name}});
+            }
+        });
+
+        const {file} = this.getAttachment(attachments[page]);
+        this.setState({
+            page,
+            attachments,
+            file,
+            isForwardDisabled: page === 0,
+            isBackDisabled: page === attachments.length - 1,
+        });
+    }
+
+    /**
+     * Increments or decrements the index to get another selected item
+     * @param {Number} deltaSlide
+    */
+    cycleThroughAttachments(deltaSlide) {
+        if ((deltaSlide < 0 && this.state.isForwardDisabled) || (deltaSlide > 0 && this.state.isBackDisabled)) {
+            return;
+        }
+
+        this.setState(({attachments, page}) => {
+            const nextIndex = page + deltaSlide;
+
+            // Check if index is near the end of the list to fetch more reports
+            if (attachments.length - nextIndex < 10) {
+                // Only fetch more if we are not already fetching so that we don't initiate duplicate requests.
+                if (this.props.report.isLoadingMoreReportActions) {
+                    return;
+                }
+
+                const sortedActions = ReportActionsUtils.getSortedReportActions(_.values(this.props.reportActions), true);
+                const oldestReportAction = _.last(sortedActions);
```

Comment:
I suspect this causes https://github.com/Expensify/App/pull/9279#issuecomment-1444513483

Recently we have 4 images/pdf, we end up loading the entire chat history which has more text.

Here the main issue 
I'm just clicking endlessly but no response (no feedback to the user)

Also, there is an image ready available to load, but we are stuck with loading old images

cc: @chiragsalian

Reply:
@Santhosh-Sellavel @chiragsalian I think I fixed this by setting up `isLoading` in the carousel